### ISSUE TITLE
dismissCallback() not call

### ISSUE THREAD

**Issue by hoangnam714** [exyte/PopupView#71]
I used the library and noticed that the dismiss Callback was not called after the popup was hidden

**Comment by f3dm76** [comment]
Hey @hoangnam714, hidden in what way? Could you please provide minimal reproducible example? Have a nice day

**Comment by hoangnam714** [comment]
here is the code i used, thank you very much       
```
 .popup(isPresented: $showDetail, type: .`default`, position: .bottom, dragToDismiss: true, closeOnTap: false, backgroundColor: .black.opacity(0.4),dismissCallback: {
            self.tabbarStore.isHident = false
            TabBarModifier.showTabBar()
        }) {
            AchievementDetailView(data: achievementData.achievements[indexAchivement], ontapDissmiss: {
                self.showDetail = false
            })
        }
```

**Comment by f3dm76** [comment]
@hoangnam714 I can not launch your coed exactly, here is the closest I could do - dismissCallback is called as expected
```
.popup(isPresented: $floats.showingTopFirst, type: .`default`, position: .bottom, dragToDismiss: true, closeOnTap: false, backgroundColor: .black.opacity(0.4), dismissCallback: {
                print("Hello")
                   }) {
                       Color.red
                   }
```

**Comment by hoangnam714** [comment]
@f3dm76 I understood why the callback function didn't work, because I didn't use the default diss miss, instead I changed the value of the state outside the code block.

**Comment by hoangnam714** [comment]
<img width="886" alt="Screen Shot 2022-07-15 at 14 54 30" src="https://user-images.githubusercontent.com/51524705/179178680-27de20b8-aa07-444c-9551-9f5abc8bb5d6.png">

**Comment by f3dm76** [comment]
I am not sure if dismiss block should be called in this case... I'll think about it. Have a nice day!

**Comment by hoangnam714** [comment]
Thank you, I will think more about this, because sometimes we don't close the popup as soon as the user touches it, instead it needs a close button for example. :)

**Comment by f3dm76** [comment]
Hey @hoangnam714, I gave it more thought and since .sheet does it, I also added this. I also added dismiss source as a parameter in dismiss callback, so you can distinguish if you need this.This feature is implemented in version 1.2.0, please give it a go. Have a marvelous day!
